src/transformers/models/mask2former/image_processing_mask2former.py

```diff
@@ -0,0 +1,1130 @@
+            ImageNet std.
+        ignore_index (`int`, *optional*):
+            Label to be assigned to background pixels in segmentation maps. If provided, segmentation map pixels
+            denoted with 0 (background) will be replaced with `ignore_index`.
+        reduce_labels (`bool`, *optional*, defaults to `False`):
+            Whether or not to decrement all label values of segmentation maps by 1. Usually used for datasets where 0
+            is used for background, and background itself is not included in all classes of a dataset (e.g. ADE20k).
+            The background label will be replaced by `ignore_index`.
+
+    """
+
+    model_input_names = ["pixel_values", "pixel_mask"]
+
+    def __init__(
+        self,
+        do_resize: bool = True,
+        size: Dict[str, int] = None,
+        size_divisor: int = 32,
+        resample: PILImageResampling = PILImageResampling.BILINEAR,
+        do_rescale: bool = True,
+        rescale_factor: float = 1 / 255,
+        do_normalize: bool = True,
+        image_mean: Union[float, List[float]] = None,
+        image_std: Union[float, List[float]] = None,
+        ignore_index: Optional[int] = None,
+        reduce_labels: bool = False,
+        **kwargs
+    ):
+        if "size_divisibility" in kwargs:
+            warnings.warn(
+                "The `size_divisibility` argument is deprecated and will be removed in v4.27. Please use "
+                "`size_divisibility` instead.",
+                FutureWarning,
+            )
+            size_divisor = kwargs.pop("size_divisibility")
+        if "max_size" in kwargs:
+            warnings.warn(
+                "The `max_size` argument is deprecated and will be removed in v4.27. Please use size['longest_edge']"
+                " instead.",
+                FutureWarning,
+            )
+            # We make max_size a private attribute so we can pass it as a default value in the preprocess method whilst
+            # `size` can still be pass in as an int
+            self._max_size = kwargs.pop("max_size")
+        else:
+            self._max_size = 1333
+
+        size = size if size is not None else {"shortest_edge": 800, "longest_edge": self._max_size}
+        size = get_size_dict(size, max_size=self._max_size, default_to_square=False)
+
+        super().__init__(**kwargs)
+        self.do_resize = do_resize
+        self.size = size
+        self.resample = resample
+        self.size_divisor = size_divisor
+        self.do_rescale = do_rescale
+        self.rescale_factor = rescale_factor
+        self.do_normalize = do_normalize
+        self.image_mean = image_mean if image_mean is not None else IMAGENET_DEFAULT_MEAN
+        self.image_std = image_std if image_std is not None else IMAGENET_DEFAULT_STD
+        self.ignore_index = ignore_index
+        self.reduce_labels = reduce_labels
+
+    @property
+    def size_divisibility(self):
```

Comment:
This function and all below seem to be copied from MaskFormer, so please add Copied from wherever possible

Reply:
Sure, I think we missed this earlier. I'll fix it now. Thanks for pointing out.

Reply:
After offline discussion with @alaradirik, it seems there's no need for a new Mask2FormerImageProcessor.

Instead, people can just use `AutoImageProcessor`, which will load a `MaskFormerImageProcessor` behind the scenes thanks to the mapping in image_processing_auto.py (which will need to be updated)

Reply:
oh okay sure...makes sense. Let's do that then.

Reply:
Yes, I will push the change shortly!

Reply:
You can link to the image processor used in the documentation. We have many models using the image processor/tokenizer/feature extractor of another model and it has never been a problem for adoption. If the image processor is the exact same as maskformer, there shouldn't be a new `Mask2FormerImageProcessor`.

Reply:
Sounds good, I'm removing Mask2FormerImageProcessor then.

Reply:
@sgugger @NielsRogge I removed the Mask2FormerImageProcessor class and updated the uploaded model repos.

Reply:
Thank you!